lambdas/process-journey-event/src/main/java/uk/gov/di/ipv/core/processjourneyevent/statemachine/stepresponses/ProcessStepResponse.java

```diff
@@ -3,26 +3,53 @@
 import lombok.AllArgsConstructor;
 import lombok.Data;
 import lombok.NoArgsConstructor;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.message.StringMapMessage;
+import uk.gov.di.ipv.core.processjourneyevent.statemachine.exceptions.StepResponseException;
 
+import java.net.URI;
+import java.net.URISyntaxException;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.Objects;
+
+import static uk.gov.di.ipv.core.library.helpers.LogHelper.LogField.LOG_CONTEXT;
+import static uk.gov.di.ipv.core.library.helpers.LogHelper.LogField.LOG_MESSAGE_DESCRIPTION;
 
 @Data
 @NoArgsConstructor
 @AllArgsConstructor
 public class ProcessStepResponse implements StepResponse {
+    private static final Logger LOGGER = LogManager.getLogger();
     private static final String JOURNEY = "journey";
     private static final String LAMBDA_INPUT = "lambdaInput";
     private static final String JOURNEY_TEMPLATE = "/journey/%s";
     private String lambda;
     private Map<String, Object> lambdaInput;
+    private String context;
 
-    @Override
     public Map<String, Object> value() {
         HashMap<String, Object> response = new HashMap<>();
-        response.put(JOURNEY, String.format(JOURNEY_TEMPLATE, lambda));
-        response.put(LAMBDA_INPUT, lambdaInput);
 
-        return response;
+        try {
+            URIBuilder uriBuilder = new URIBuilder(String.format(JOURNEY_TEMPLATE, lambda));
```

Comment:
This is how we pass the context around the step function, since it cannot take optional parameters, so we must integrate it into the journey uri. This change also affects the string matching mechanisms in the step function and local running configs